service/r2/convert.go

```diff
@@ -34,7 +34,7 @@ type namespaceJSON struct {
 	ForRuleSetVersion int    `json:"forRuleSetVersion"`
 }
 
-func newNamespaceJSON(nv *rules.NamespaceView) namespaceJSON {
+func newNamespaceJSON(nv *models.NamespaceView) namespaceJSON {
```

Comment:
Do we need these ctors etc? Should these be replaced by what we defined in the `models` package in `m3metrics`?

Reply:
Yeah that needs to be done. I mentioned in the PR description that I am going to do that as a separate PR as there are a few things that are blocked by this being landed and if we land this first than the things that are blocked and using the new structs can be worked on in parallel.

Reply:
Cool sounds good.